### PR TITLE
Joomla Update

### PR DESCRIPTION
**Pull request by Llewellynvdm**
Update Joomla Docker documentation to improve clarity and enhance configuration options

- Changed container network configuration to use Docker networks instead of linking containers.
- Introduced new environment variables for database configuration:
  - `JOOMLA_DB_PASSWORD_FILE`: Path to a file containing the database password.
  - `JOOMLA_DB_TYPE`: Database type with options for "mysqli" and "pgsql".
- Added environment variables for auto-deployment configuration:
  - `JOOMLA_SITE_NAME`, `JOOMLA_ADMIN_USER`, `JOOMLA_ADMIN_USERNAME`, `JOOMLA_ADMIN_PASSWORD`, `JOOMLA_ADMIN_EMAIL`
  - `JOOMLA_EXTENSIONS_URLS`, `JOOMLA_EXTENSIONS_PATHS`
  - `JOOMLA_SMTP_HOST`, `JOOMLA_SMTP_HOST_PORT`
- Updated port mapping instructions to reflect the new network setup.
- Corrected the Dockerfile reference link for adding additional PHP extensions.

These changes improve the setup process, provide more flexible configuration options, and correct outdated references.

### CODE REVIEW

File: joomla/content.md [hunk 1]
```diff
@@ -9,31 +9,45 @@ Joomla is a free and open-source content management system (CMS) for publishing
 # How to use this image
 
 ```console
-$ docker run --name some-%%REPO%% --link some-mysql:mysql -d %%IMAGE%%
+$ docker run --name some-%%REPO%% --network some-network -d %%IMAGE%%
```
Comment (tianon) [inline joomla/content.md:12]: You've used `joomla-network` in the other examples and `some-network` here -- the latter is more common in other image docs, but I don't have a strong preference.  Did you want them to be consistent?

**Review by tianon — COMMENTED** [review]
Looks fine to me sans my minor nit about consistency (that isn't a blocker, to be clear), and the CI failure for formatting (which I'm happy to resolve for you if you want/need me to).

**Comment by tianon** [comment]
(Oh, I don't have commit access to your PR as you've unchecked that option, so I'm happy to walk you though the formatting fixes if you'd prefer and want the assist, but I can't make them for you directly unless you enable the checkbox on this PR that would allow me to do so.)

File: joomla/content.md (hunk 1 above)
Comment (Llewellynvdm) [reply joomla/content.md:12]: oops, yes... I overlooked that.

**Comment by Llewellynvdm** [comment]
For some reason I do/did not see that checkbox.

image

I would gladly enable the checkbox on this PR, but can't find it. So I will just pushed an update...

File: joomla/content.md [hunk 2]
```diff
@@ -9,31 +9,45 @@ Joomla is a free and open-source content management system (CMS) for publishing
...
 Then, access it via `http://localhost:8080` or `http://host-ip:8080` in a browser.
 
-If you'd like to use an external database instead of a linked `mysql` container, specify the hostname and port with `JOOMLA_DB_HOST` along with the password in `JOOMLA_DB_PASSWORD` and the username in `JOOMLA_DB_USER` (if it is something other than `root`):
+If you'd like to use an external database instead of a MySQL container, specify the hostname and port with `JOOMLA_DB_HOST` along with the password in `JOOMLA_DB_PASSWORD` and the username in `JOOMLA_DB_USER` (if it is something other than `root`):
 
 ```console
-$ docker run --name some-%%REPO%% -e JOOMLA_DB_HOST=10.1.2.3:3306 \
-    -e JOOMLA_DB_USER=... -e JOOMLA_DB_PASSWORD=... -d %%REPO%%
+$ docker run --name some-%%REPO%% --network some-network -e JOOMLA_DB_HOST=10.1.2.3:3306 \
+    -e JOOMLA_DB_USER=... -e JOOMLA_DB_PASSWORD=... -d %%IMAGE%%
```
Comment (Llewellynvdm) [inline joomla/content.md:50]: Should this be `%%REPO%%` or `%%IMAGE%%`

File: joomla/stack.yml [hunk 3]
```diff
@@ -1,19 +1,41 @@
...
+      JOOMLA_DB_NAME: joomla_db
+      JOOMLA_SITE_NAME: Joomla
+      JOOMLA_ADMIN_USER: Joomla Hero
+      JOOMLA_ADMIN_USERNAME: joomla
+      JOOMLA_ADMIN_PASSWORD: joomla@secured
+      JOOMLA_ADMIN_EMAIL: joomla@example.com
+    volumes:
+      - joomla_data:/var/www/html
+    networks:
+      - joomla_network
```
Comment (Llewellynvdm) [inline joomla/stack.yml:21]: Should I also change this to `some-network` or is this okay?

File: joomla/content.md (hunk 2 above)
Comment (tianon) [reply joomla/content.md:50]: This one should be `%%IMAGE%%` and is an excellent catch we clearly missed previously! :smile:

File: joomla/stack.yml (hunk 3 above)
Comment (tianon) [reply joomla/stack.yml:21]: I think this is probably fine since it's self-contained in an example file that does create the network also.

**Comment by tianon** [comment]
> I would gladly enable the checkbox on this PR, but can't find it.

:sob: I guess maybe there's a setting on the `joomla-docker` org that's preventing it?  Not sure :disappointed: :heart:

**Review by tianon — APPROVED** [review]

**Comment by tianon** [comment]
Thank you!

**Pull request merged**